Add Fry/DEA App

nathanielbuck opened the pull request:
## Description
Add a new form application for Fry/DEA.

Sibling PR: https://github.com/department-of-veterans-affairs/content-build/pull/1159

nathanielbuck committed:
Add Fry/DEA App

nathanielbuck committed:
Add Fry/DEA App

cypress-viewport-updater-bot[bot] committed:
Update Cypress Viewport Data (Automatic Update) (#21252)

* update cypress.json

* update viewportPreset.js

Co-authored-by: cypress-viewport-updater-bot[bot] <77450695+cypress-viewport-updater-bot[bot]@users.noreply.github.com>

theodur reviewed (commented):
Please feel free to add this app to the [allow-list](https://github.com/department-of-veterans-affairs/vets-website/blob/main/config/changed-apps-build.json), so it can take advantage of faster CI times.

Also, it looks like there's  a failed unit test related to your app.

ddzz committed:
Update references to QA Standards data repo in CI (#21241)

mdewey committed:
(feature) added some accessbility changes (#21255)

* (feature) added some accessbility changes

* (feature) changed link

* (feature) updated descibred by

* (tests) added minial testing for accessbility

tmitchellgcio committed:
update alertbox to va-alerts, add proptypes, update loading (#21256)

* update alertbox to va-alerts, add proptypes, update loading

* fix unit tests

* change to array for facilities

bkjohnson committed:
Bump component-library for TextInput & Select fixes (#21245)

* Bump component-library to prerelease version

* Update to full release

ddzz committed:
Refactor My VA Dashboard E2E tests (#21260)

ryguyk committed:
VACMS-9263: Sends user to clicked search result page even if click tracking errors. (#21248)

holdenhinkle committed:
update PR description (#21266)

ddzz committed:
Fix certificate issue in E2E proxy test (#21239)

jaredcunha committed:
Fix link text for MailMessage (#21268)

maxx1128 committed:
VA-9320: Fix bug if static COVID data is unavailable (#21267)

* VA-9161: Fix bug if static COVID data is unavailable

* VA-9161: Change static data to use base endpoint

yonidoronpeters committed:
DHP-277 Use reactive button to toggle login modal (#21236)

* DHP-277 Use reactive button to toggle login modal

* DHP-277 Fix sign in modal button by
1. Fixing props in functional component
2. Fixing import for UnauthenticatedPageContent in App.jsx

* DHP-277 Add and fix tests for "Sign in or create an account" button

* DHP-277 Small styling fix

Dr-Pongo committed:
Updated otherVA link destinations with new URLs (#21272)

theodur committed:
Replace eslint-plugin-va with @department-of-veterans-affairs/eslint-plugin (#21259)

* Use ESLint plugin package

* Update rule overrides

* Update eslint-plugin-va references

* Remove eslint-config-vagov

* Remove eslint-plugin-va

* Remove eslint-plugin-va from package.json

reblg committed:
Dhp 223 FAQs (#21274)

* DHP-223 Add FAQs content

Co-authored-by: Rebecca Green <rebecca.green@thoughtworks.com>
Co-authored-by: Yoni Doron-Peters <jonathan.d@thoughtworks.com>

dan-brooking committed:
Update housing rate on card for spouse/child chapter 35. (#21284)

* Update housing rate on card for spouse/child chapter 35.

* Adding env flag.

jtmst committed:
8870 added health connect number to facility locator search results. … (#21212)

* 8870 added health connect number to facility locator search results. removed hard coded data

* 8870 removed unnecessary return line

* added proptypes for resultList component.  code cleanup

* eslint cleanup

* 8870 prop list order fixed

* refactor resultsList

* 8870 change requests

* 8870 change request addressed

* 8870 lint cleanup

dan-brooking committed:
Removing prod flag. (#21285)

allisonlu committed:
feature/VACMS-8340 VA.gov Homepage redesign iteration (#21061)

* VACMS-8359:
- set up stylesheet specifically for blog promo on homepage
- fixed image size and positioning

* VACMS-8359:
- styles for the horizontal rule (we can delete this if we get this into the design system)

* VACMS-8359:
- fixed image and container height on tablet and small screens

* VACMS-8359: reduced width of horizontal rule on mobile screens

* VACMS-8359: removed styles for class that doesn't belong in design system

* VACMS-8360 Homepage hero component functionality for random images (#20790)

* VACMS-8360 set up architecture for home page hero React widget

* VACMS-8360
- created a component for the Veteran images in the hero with logic that randomizes the order on pageload

* VACMS-8360:
- created scss stylesheet for homepage hero
- added some styles for positioning

* VACMS-8360:
- background a gradient background color that goes off the grid
- added background colors to each image

* VACMS-8360:
- added documentation to stylesheet
- added gradient background color that goes off the grid
- centered images better in container

* VACMS-8360:
- added styles for button

* VACMS-8360 changed height to min-height for Safari

* VACMS-8360:
- fixed eslint errors
- refactored imagePaths array

* VACMS-8360:
- added documentation
- removed Object.keys(...)

* VACMS-8360:
- moved scss import to file where widget is mounted
- moved documentation block

* VACMS-8361: Homepage - Search + Common Tasks  (#21008)

* VACMS-8361:
- created a stylesheet for Common Tasks section
- added initial styles for colors and positions

* VACMS-8361:
- removed font-size for links for now (usability testing purposes)
- added styles for columns within a list
- added date + author info

* VACMS-8361:
- tweaked Search input + button field width
- made linear-gradient background more precise

* VACMS-8361:
- removed some padding and spacing styles

* VACMS-8361: clean up

* VACMS-8362 Homepage redesign - Search functionality (#21136)

* VACMS-8362:
- search up infrastructure for Search functionality on homepage
- created HomepageSearch component

* VACMS-8362:
- implement Search input component (from design system) with Typeahead for Homepage Common Tasks section (midpage)

* VACMS-8362:
- removed mock suggestions, trying to use Typeahead API instead
- added documentation

* VACMS-8362:
- implemented Typeahead logic with Search Input component from design system
- polished functionality when input changes
- removed mock suggestions
- (calling from prod API for demo and testing purposes)

* VACMS-8362:
- updated Typeahead API endpoint
- added some documentation

* VACMS-8362:
- removed unnecessary styles

* VACMS-8362:
- removed unnecessary component props
- removed state setter from fetch function
- moved return statement to inside of catch block
- replaced setTimeOut with an await Promise

* VACMS-8340: cleaned up eslint errors

* revert irrelevant changes (that somehow creeped into this branch, probably due to eslint autofix?)

brianseek committed:
Added general developer readme to check-in root. (#21281)

* Added general developer readme to check-in root.

* Fixed typos and added details.

* Suggested changes from review.

* Update src/applications/check-in/README.md

Co-authored-by: Adrian Rollett <acrollet@gmail.com>

* Apply suggestions from code review

Co-authored-by: Adrian Rollett <acrollet@gmail.com>

* Added screenshot size specifics and UUID section.

* Force CI re-run

* Force CI re-run

Co-authored-by: Adrian Rollett <acrollet@gmail.com>
Co-authored-by: Darius Dzien <dariusz.dzien@adhocteam.us>

brianseek committed:
fix for Cannot read properties of undefined (reading 'startTime') (#21265)

* Prevented rendering of confirmation page if no appointment and routed back to list with refresh.

* Added jumpToPage to useEffect dep array.

theodur committed:
Remove content-build files from .gitignore (#21257)

mdewey committed:
Profile/feature/42144/force bad address (#21276)

* (feature) added new feature toggle

* (feature) added logic to force showing and hideingof bad address for test purposes

* (feature) turned off force bad address by default

* (test) added e2e tests for the forcing of BAI

asg5704 committed:
Creates OAuth authorize endpoint code (#21273)

* initial commit

* creates oauth/utils unit test

* removes unnecssary comments; remove global on window object

* updates sha256 fn; updates utilities

* updates sha256 and urlencode

* removes alexOptions from oauth/utils; removes oAuthConfig fn

* updates the usip config; updates unit test for oauth/utils

* updates client_id

* updates redirect_uri

* updates redirect_uri

FernandoBoza committed:
added UI for remarriage additional considerations. (#21294)

Co-authored-by: fer.boza <fernando.boza@accenturefederal.com>

lzapson committed:
added conditional so users can complete form as they started (#21292)

scottymeyers committed:
utilizes va-process-list component on COE intro (#21291)

Mottie committed:
HLR: Remove deprecated v1 code (#21261)

* HLR: Remove v1 code

* Update mock data & tests

* Remove wizard legacy question & e2e tests

* Wizard: derp restore & rename legacyNo question

* Code cleanup

* Fix unit tests

brdunfield committed:
[Check-in] View Mixed Language disclaimer on non-english pages (#21244)

* Add MixedLanguageDisclaimer component and feature toggle
* Add wrapper component to all active pages

Commit:
41319 preferred name error rework (#21290)

* (feature) add optional server delay

* (fix) require preferred name if submitted blank

* (test) update e2e for preferred name edge case

* revert unneeded changes

acrollet committed:
Format mock checkInWindow times to match production (#21289)

* Format mock checkInWindow times to match production

* Fix e2e test

brdunfield committed:
Update mixed language disclaimer style (#21295)

dsrr47 committed:
added proptypes, added logic to disable dropdown value (#21296)

Co-authored-by: David <dsrr74@gmail.com>

brdunfield committed:
content update for mixed language disclaimer (#21302)

khathuynh-tw committed:
Va virtual agent 506 june chatbot copy changes (#21301)

* [department-of-veterans-affairs/va-virtual-agent#506] Update disclaimer

* [department-of-veterans-affairs/va-virtual-agent#506] Remove references to virtual agent on chatbot page

* [department-of-veterans-affairs/va-virtual-agent#506] Update chatbot masthead

dsrr47 committed:
benefit defaults to ch33 anytime military status changes. (#21298)

Co-authored-by: David <dsrr74@gmail.com>

OptionSelect committed:
41484 - [CDP] [MCP] Update medical-copays Cypress tests under CDP (#21300)

* First two cypress test files passing fully now, committing to save progress

* E2E tests all pass, begin work on unit tests

* Update MCP tests to use proper page elements again

yonidoronpeters committed:
DHP-223 Address PR comment about generic id's for FAQ components (#21304)

smeneni committed:
Fixed text errors

timwright12 committed:
Skip appeals lighthouse tests (#21306)

OptionSelect committed:
Updated component to not open in new page and added requested download attribute (#21307)

simiadebowale committed:
Remove Type of Care label on TMP appointments. (#21264)

* remove Type of care label if serviceType is not provided

* udated logic

dsrr47 committed:
Remove flags 42475 and 42472 (#21309)

* benefit defaults to ch33 anytime military status changes.

* removed env variable

Co-authored-by: David <dsrr74@gmail.com>

vbahinwillit committed:
39624-eligibility-check-using-wrong-api-call-when-vaos-flag-is-on (#21194)

* modified to use vaos api call

* modified unit tests

* fix e2e test error

* fix unit test

ddzz committed:
Exclude Cypress lint rules in unit tests (#21313)

* test

* Exclude Cypress lint rules in unit tests

* Revert test

Mottie committed:
COE: Add mock data (#21299)

* Add minimal/maximal data

* Update cypress test

brdunfield committed:
Add telephone appointment feature toggle for check-in application (#21310)

acrollet committed:
Add appt kind to pre-check-in mocks (#21314)

smeneni committed:
fix: fixed alert text error

smeneni committed:
Merge branch 'feature/fry-dea-add-app' of https://github.com/department-of-veterans-affairs/vets-website into feature/fry-dea-add-app

asg5704 reviewed (approved):
No impact on my files. LGTM

bkjohnson commented on src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:
```diff
@@ -160,7 +160,7 @@ class SaveInProgressIntro extends React.Component {
           );
         }
-      } else if (prefillAvailable && !verifiedPrefillAlert) {
+      } else if (!prefillAvailable && !verifiedPrefillAlert) {
```
Why are things changing here?

nathanielbuck replied:
@smeneni, this isn't the correct place to make these changes. In general, we should not be making changes to platform code.

micahchiang reviewed (changes requested):
This pull request is big, even for one that adds a new application. Was main properly merged into this branch before submitting this PR? Looks like a ton of files that have nothing to do with the DEA app have been included.

nathanielbuck commented:
I'm not exactly sure what happened after my initial commit (I've been away for the past week). I think @smeneni may have pushed a bad merge/rebase. I'm going to close this and open a fresh PR.
